src/transformers/models/idefics3/configuration_idefics3.py

```diff
@@ -0,0 +1,222 @@
+            )
+
+        return cls.from_dict(config_dict, **kwargs)
+
+
+
+class Idefics3Config(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`Idefics3Model`]. It is used to instantiate a
+    Idefics3 model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the model of the Idefics3
+    [HuggingFaceM4/idefics3-8b](https://huggingface.co/HuggingFaceM4/idefics3-8b) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Args:
+        use_cache (`bool`, *optional*, defaults to `True`):
+            Whether or not the model should cache the key/value pairs of the attention mechanism.
+        image_token_id (`int`, *optional*, defaults to 128257):
+            The id of the "image" token.
+        tie_word_embeddings (`bool`, *optional*, defaults to `False`):
+            Whether or not to tie the word embeddings with the token embeddings.
+        vision_config (`IdeficsVisionConfig` or `dict`, *optional*):
+            Custom vision config or dict
+        text_config (`LlamaConfig` or `dict`, *optional*):
+            Custom text config or dict for the text model
+        scale_factor (`int`, *optional*, defaults to 2):
+            The scale factor for the image encoder.
+        vocab_size (`int`, *optional*, defaults to 100000):
+            The size of the vocabulary.
+        hidden_size (`int`, *optional*, defaults to 4096):
+            Dimensionality of the encoder layers and the pooler layer.
+        pad_token_id (`int`, *optional*, defaults to 128002):
+            The id of the padding token.
+        max_position_embeddings (`int`, *optional*, defaults to 131072):
+            The maximum length of the input sequence.
+
+    Example:
+    ```python
+    >>> from transformers import Idefics3Model, Idefics3Config
+    >>> # Initializing configuration
+    >>> configuration = Idefics3Config()
+    >>> # Initializing a model from the configuration
+    >>> model = Idefics3Model(configuration)
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "idefics3"
+    is_composition = True
+
+    def __init__(
+        self,
+        use_cache=True,
+        image_token_id=128257,
+        tie_word_embeddings=False,
+        vision_config=None,
+        text_config=None,
+        scale_factor=2,
+        vocab_size=100000,
+        hidden_size=4096,
+        pad_token_id=128_002,
+        max_position_embeddings=131_072,
+        **kwargs,
+    ):
+        self.image_token_id = image_token_id
+        self.use_cache = use_cache
+        self.tie_word_embeddings = tie_word_embeddings
+        self.vocab_size = vocab_size
```

Comment:
Is it possible to rely on `text_config.vocab_size`? We deprecated `vocab_size` in all VLMs and are trying to rely on `text_config`?

Reply:
thanks! I'll remove the vocab size from the main model and rely on the parm of the text_config.

Reply:
Done and checked that everything still works fine 👍